internal/storage/gcs/folder.go

```diff
@@ -15,11 +15,34 @@
 package gcs
 
 import (
+	"strings"
 	"time"
+
+	"cloud.google.com/go/storage/control/apiv2/controlpb"
 )
 
 type Folder struct {
 	Name           string
 	Metageneration int64
 	UpdateTime     time.Time
 }
+
+func ConvertToGCSFolderFromGetFolderResponse(bucketName string, attrs *controlpb.Folder) *Folder {
+	// Setting the parameters in Folder and doing conversions as necessary.
+	return &Folder{
+		Name:           findFolderName(bucketName, attrs.Name),
+		Metageneration: attrs.Metageneration,
+		UpdateTime:     attrs.GetUpdateTime().AsTime(),
+	}
+}
+
+// In HNS, folder paths returned by control client APIs are in the form of:
+//
+// projects/_/buckets/{bucket}/folders/{folder}
+// This method extracts the folder name from such a string.
+func findFolderName(bucketName, object string) string {
```

Comment:
also "object string" could be "object fullPath" or something signifying what actually needs to be passed.

Reply:
It will unnecessary increase the type conversion. I can change object with fullPath name.